File: Behavioral/ChainOfResponsibility/Java/Supervisor.java
// Supervisor.java

public class Supervisor extends SupportHandler {
  @Override
  public void handleRequest(String request) {
    if (request.equals("Intermediate Issue")) {
      System.out.println("Supervisor: Handling request - " + request);
    } else if (nextHandler != null) {
      nextHandler.handleRequest(request); // Only pass if not handled
    } else {
      System.out.println("No handler available for - " + request);
    }
  }
}

File: Behavioral/ChainOfResponsibility/Python/supervisor.py
# supervisor.py

from support_handler import SupportHandler

class Supervisor(SupportHandler):
  def handle_request(self, request):
    if request == "Intermediate Issue":
      print(f"Supervisor: Handling request - {request}")
    elif self._next_handler:
      self._next_handler.handle_request(request)
    else:
      print(f"No handler available for - {request}")
